pkg/apis/serving/v1/configuration_validation.go

```diff
@@ -71,11 +71,12 @@ func (csf *ConfigurationStatusFields) Validate(ctx context.Context) *apis.FieldE
 }
 
 // validateLabels function validates configuration labels
+// Any label with prefix serving.knative.dev is not allowed except
+// serving.knative.dev/visibility , serving.knative.dev/service, serving.knative.dev/route label.
 func (c *Configuration) validateLabels() (errs *apis.FieldError) {
 	for key, val := range c.GetLabels() {
 		switch {
 		case key == serving.VisibilityLabelKey:
-			errs = errs.Also(validateClusterVisibilityLabel(val))
```

Comment:
❓ Is there a reason why the configuration was/is allowed to have this label?

For the Service I understand it'll propagate to the route. Having it on the configuration means I could have a referencing route with a different visibility?

Reply:
When creating a ksvc, the configuration object created gets the labels from the ksvc as well. Since ksvc can have visibility, it'd cause the configuration to be rejected.

Perhaps as a separate issue, can have it only pass the relevant labels down to configurations.